index.js

```diff
@@ -140,37 +189,98 @@ function loader (registryOrVersion) {
+          name: typeof ench.id === 'string' ? ench.id.replace(/minecraft:/, '') : null
+        }))
       }
       throw new Error("Don't know how to get the enchants from an item on this mc version")
     }
 
     set enchants (normalizedEnchArray) {
-      const isBook = this.name === 'enchanted_book'
-      const enchListName = registry.supportFeature('nbtNameForEnchant')
-      const type = registry.supportFeature('typeOfValueForEnchantLevel')
-      if (type === null) throw new Error("Don't know the serialized type for enchant level")
-      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      if (registry.type === 'bedrock') {
+        const enchs = normalizedEnchArray.map(({ name, lvl }) => ({
+          id: { type: 'short', value: registry.enchantmentsByName[name].id },
+          lvl: { type: 'short', value: lvl }
+        }))
+
+        if (enchs.length !== 0) {
+          if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+          this.nbt.value.ench = { type: 'list', value: { type: 'compound', value: enchs } }
+        }
+      } else if (registry.type === 'pc') {
+        const isBook = this.name === 'enchanted_book'
+        const enchListName = registry.supportFeature('nbtNameForEnchant')
+        const type = registry.supportFeature('typeOfValueForEnchantLevel')
+        if (type === null) throw new Error("Don't know the serialized type for enchant level")
+        if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+
+        const enchs = normalizedEnchArray.map(({ name, lvl }) => {
+          const value =
+            type === 'short'
+              ? registry.enchantmentsByName[name].id
+              : `minecraft:${registry.enchantmentsByName[name].name}`
+          return { id: { type, value }, lvl: { type: 'short', value: lvl } }
+        })
+
+        if (enchs.length !== 0) {
+          this.nbt.value[isBook ? 'StoredEnchantments' : enchListName] = {
+            type: 'list',
+            value: { type: 'compound', value: enchs }
+          }
+        }
 
-      const enchs = normalizedEnchArray.map(({ name, lvl }) => {
-        const value = type === 'short' ? registry.enchantmentsByName[name].id : `minecraft:${registry.enchantmentsByName[name].name}`
-        return { id: { type, value }, lvl: { type: 'short', value: lvl } }
-      })
+        // The 'registry.itemsByName[this.name].maxDurability' checks to see if this item can lose durability
+        if (
+          registry.supportFeature('whereDurabilityIsSerialized') === 'Damage' &&
+          registry.itemsByName[this.name].maxDurability
+        ) {
+          this.nbt.value.Damage = { type: 'int', value: 0 }
+        }
+      }
+    }
+
+    // TODO: namespaces other than minecraft:
+    // The item palette on bedrock can contain items
+    // with other namespaces, probably need to handle
+    // this in prismarine-registry
+    // Also this needs to be tested
+    get canPlaceOn () {
```

Comment:
canPlaceOn -> blocksCanPlaceOn
canDestroy -> blocksCanDestroy

Also must be added to doc that this applies to adventure mode.
```js
this.nbt.value.CanPlaceOn = nbt.list(nbt.string( blocks.map(b => b.startsWith('minecraft:') ? b : 'minecraft:' + b ) ))
```